resctrl/utils.go

```diff
@@ -190,20 +212,16 @@ func getPids(containerName string) ([]int, error) {
 // getAllProcessThreads obtains all available processes from directory.
 // e.g. ls /proc/4215/task/ -> 4215, 4216, 4217, 4218
 // func will return [4215, 4216, 4217, 4218].
-func getAllProcessThreads(path string) ([]int, error) {
-	processThreads := make([]int, 0)
+func getAllProcessThreads(path string) ([]string, error) {
+	processThreads := make([]string, 0)
 
 	threadDirs, err := ioutil.ReadDir(path)
 	if err != nil {
 		return processThreads, err
 	}
 
 	for _, dir := range threadDirs {
-		pid, err := strconv.Atoi(dir.Name())
-		if err != nil {
-			return nil, fmt.Errorf("couldn't parse %q dir: %v", dir.Name(), err)
-		}
-		processThreads = append(processThreads, pid)
+		processThreads = append(processThreads, dir.Name())
```

Comment:
pids used to be ints and were passed to the external `ps` call eventually. Since `ps` is replaced by `libcontainer` function call which accepts strings, we no longer need this additional `Atoi` conversion.